pydantic/main.py

```diff
@@ -365,29 +367,45 @@ def copy(
 
         return self._copy_and_set_values(values, fields_set, deep=deep)
 
-    @classmethod
-    def model_json_schema(
-        cls, by_alias: bool = True, ref_template: str = default_ref_template
-    ) -> typing.Dict[str, Any]:
-        cached = cls.__schema_cache__.get((by_alias, ref_template))
-        if cached is not None:
-            return cached
-        s = model_schema(cls, by_alias=by_alias, ref_template=ref_template)
-        cls.__schema_cache__[(by_alias, ref_template)] = s
-        return s
+    # @classmethod
+    # def model_json_schema(
+    #     cls, by_alias: bool = True, ref_template: str = default_ref_template
+    # ) -> typing.Dict[str, Any]:
+    #     cached = cls.__schema_cache__.get((by_alias, ref_template))
+    #     if cached is not None:
+    #         return cached
+    #     s = model_schema(cls, by_alias=by_alias, ref_template=ref_template)
+    #     cls.__schema_cache__[(by_alias, ref_template)] = s
+    #     return s
+
+    # @classmethod
+    # def schema_json(
+    #     cls, *, by_alias: bool = True, ref_template: str = default_ref_template, **dumps_kwargs: Any
+    # ) -> str:
+    #     from .json import pydantic_encoder
+
+    #     return cls.__config__.json_dumps(
+    #         cls.model_json_schema(by_alias=by_alias, ref_template=ref_template),
+    #         default=pydantic_encoder,
+    #         **dumps_kwargs,
+    #     )
 
     @classmethod
-    def schema_json(
-        cls, *, by_alias: bool = True, ref_template: str = default_ref_template, **dumps_kwargs: Any
-    ) -> str:
-        from .json import pydantic_encoder
+    def json_schema(cls, *, ref_template=None):
```

Comment:
I'll review more tomorrow, but just one quick note now: `json_schema` has been renamed to `model_json_schema`, see #4672